src/gui/csvImport/CsvImportWidget.cpp

```diff
@@ -0,0 +1,289 @@
+
+    connect(m_ui->buttonBox, SIGNAL(accepted()), this, SLOT(writeDatabase()));
+    connect(m_ui->buttonBox, SIGNAL(rejected()), this, SLOT(reject()));
+}
+
+void CsvImportWidget::comboChanged(int comboId) {
+    QComboBox* currentSender = qobject_cast<QComboBox*>(m_comboMapper->mapping(comboId));
+    if (currentSender->currentIndex() != -1) {
+        //here is the line that actually updates the GUI table
+        m_parserModel->mapColumns(currentSender->currentIndex(), comboId);
+    }
+    updateTableview();
+}
+
+void CsvImportWidget::skippedChanged(int rows) {
+    m_parserModel->setSkippedRows(rows);
+    updateTableview();
+}
+
+CsvImportWidget::~CsvImportWidget() {}
+
+void CsvImportWidget::configParser() {
+    m_parserModel->setBackslashSyntax(m_ui->checkBoxBackslash->isChecked());
+    m_parserModel->setComment(m_ui->comboBoxComment->currentText().at(0));
+    m_parserModel->setTextQualifier(m_ui->comboBoxTextQualifier->currentText().at(0));
+    m_parserModel->setCodec(m_ui->comboBoxCodec->currentText());
+    m_parserModel->setFieldSeparator(m_ui->comboBoxFieldSeparator->currentText().at(0));
+}
+
+void CsvImportWidget::updateTableview() {
+    m_ui->tableViewFields->resizeRowsToContents();
+    m_ui->tableViewFields->resizeColumnsToContents();
+
+    for (int c=0; c<m_ui->tableViewFields->horizontalHeader()->count(); ++c) {
+        m_ui->tableViewFields->horizontalHeader()->setSectionResizeMode(
+                    c, QHeaderView::Stretch);
+    }
+}
+
+void CsvImportWidget::updatePreview() {
+
+    m_ui->labelSizeRowsCols->setText(m_parserModel->getFileInfo());
+    m_ui->spinBoxSkip->setValue(0);
+    m_ui->spinBoxSkip->setMaximum(m_parserModel->rowCount()-1);
+
+    int i;
+    QStringList list(tr("Not present in CSV file"));
+
+    for (i=1; i<m_parserModel->getCsvCols(); i++) {
+        QString s = QString(tr("Column ")) + QString::number(i);
+        list << s;
+    }
+    m_comboModel->setStringList(list);
+
+    i=1;
+    Q_FOREACH (QComboBox* b, m_combos) {
+        if (i < m_parserModel->getCsvCols()) {
+            b->setCurrentIndex(i);
+        }
+        else {
+            b->setCurrentIndex(0);
+        }
+        ++i;
+    }
+}
+
+void CsvImportWidget::load(const QString& filename, Database* const db) {
+    m_db = db;
+    m_parserModel->setFilename(filename);
+    m_ui->labelFilename->setText(filename);
+    Group* group = m_db->rootGroup();
+    group->setUuid(Uuid::random());
+    group->setNotes(tr("Imported from CSV file\nOriginal data: ") + filename);
+
+    parse();
+}
+
+void CsvImportWidget::parse() {
+    configParser();
+    QApplication::setOverrideCursor(Qt::WaitCursor);
+    bool good = m_parserModel->parse();
+    QApplication::restoreOverrideCursor();
+    updatePreview();
+    m_ui->pushButtonWarnings->setEnabled(!good);
+}
+
+void CsvImportWidget::showReport() {
+    MessageBox::warning(this, tr("Syntax error"), tr("While parsing file...\n")
```

Comment:
Well, you shouldn't show a gazillion error messages at once anyway. A message box that extends beyond the edge of the screen and therefore has no usable OK button isn't any better.
You should only show the first few message and then add "<N> more messages" if there are any errors left.